src/js/components/SharedComponents/navigation/UserButton.jsx

```diff
@@ -42,16 +48,45 @@ export default class UserButton extends React.Component {
             hideDropdown = "";
         }
 
+        let settingsButton = null;
+        let displaySettings = false;
+        if (this.props.user.website_admin) {
+            displaySettings = true;
+        }
+        else {
+            for (let i = 0; i < this.props.user.affiliations.length; i++) {
```

Comment:
Recommend moving this logic to `permissionsHelper.js` so it can be unit tested and reused if needed.